google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java

```diff
@@ -316,16 +316,21 @@ public void connect(@NotNull ConnectionCallback<AppEngineDeploymentConfiguration
       ServerRuntimeInstance<AppEngineDeploymentConfiguration> {
 
     private AppEngineServerConfiguration configuration;
+    private AppEngineAction currentAction;
 
     public AppEngineRuntimeInstance(
         AppEngineServerConfiguration configuration) {
       this.configuration = configuration;
     }
 
     @Override
-    public void deploy(@NotNull final DeploymentTask<AppEngineDeploymentConfiguration> task,
```

Comment:
isn't an AppEngineRuntimeInstance scoped to a deployment? is synchronization necessary?